src/apps/cardano/address.py

```diff
@@ -16,12 +25,27 @@ def derive_address_and_node(keychain, path: list):
     address_data = [address_root, address_attributes, address_type]
     address_data_encoded = cbor.encode(address_data)
 
-    address = base58.encode(
-        cbor.encode(
-            [cbor.Tagged(24, address_data_encoded), crc.crc32(address_data_encoded)]
-        )
-    )
-    return (address, node)
+    return (_encode_address_raw(address_data_encoded), node)
+
+
+def validate_address(address) -> bool:
```

Comment:
I am not sure I like `validate_` prefix. Probably `is_valid_` is better but I can't find a good wording for it.
Also, this *does* not say what is being validated. What we validate is *only* outer CBOR encoding, not full validity.
Maybe `address_has_valid_root_cbor_structure` ?